src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php

```diff
@@ -73,12 +70,9 @@ public function current()
     }
 
     /**
-     * @return \RecursiveIterator
-     *
      * @throws AccessDeniedException
      */
-    #[\ReturnTypeWillChange]
-    public function getChildren()
+    public function getChildren(): \RecursiveIterator
```

Comment:
This change triggers the deprecation again. PHP expects the return type to be `RecursiveDirectoryIterator`, but that type is too narrow for our current implementation, unfortunately.

Reply:
Files as #42468. I'm reverting this change for now.